bundle/bundle_test.go

```diff
@@ -205,11 +208,48 @@ func TestValuesOrDefaults_Required(t *testing.T) {
 	// Example: It makes perfect sense for a user to specify --set minimum=0
 	// and in so doing meet the requirement that a value be specified.
 	vals["minimum"] = 0
-	res, err := ValuesOrDefaults(vals, b)
+	res, err := ValuesOrDefaults(vals, currentVals, b)
 	is.NoError(err)
 	is.Equal(0, res["minimum"])
 }
 
+func TestValuesOrDefaults_Immutable(t *testing.T) {
+	is := assert.New(t)
+	vals := map[string]interface{}{
+		"enabled":   true,
+		"namespace": "new-ns",
+	}
+	currentVals := map[string]interface{}{
+		"namespace": "actual-ns",
+		"enabled":   true,
+	}
+	b := &Bundle{
+		Definitions: map[string]*definition.Schema{
+			"namespaceType": {
+				Type: "string",
+			},
+			"enabledType": {
+				Type:    "boolean",
+				Default: false,
+			},
+		},
+		Parameters: &ParametersDefinition{
+			Fields: map[string]ParameterDefinition{
+				"namespace": {
+					Definition: "namespaceType",
+					Immutable:  true,
+				},
+				"enabled": {
+					Definition: "enabledType",
+				},
+			},
+		},
+	}
+
+	_, err := ValuesOrDefaults(vals, currentVals, b)
+	is.Error(err)
```

Comment:
What about testing the returned error is due to immutability? With something like this:
https://github.com/stretchr/testify/blob/ffdc059bfe9ce6a4e144ba849dbedead332c6053/assert/assertion_forward.go#L137 ?

Reply:
Agreed, checking the error returned would be good

Reply:
Indeed. I fixed it.